frontends/web/src/contexts/ConfigContext.tsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2021 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createContext, Dispatch, SetStateAction, useEffect, useRef, useState } from 'react';
+import { Config, getConfig, setConfig as postConfig } from '../api/config';
+
+export interface ConfigContextInterface {
+    config: Config | undefined
+    setConfig: Dispatch<SetStateAction<Config>>
+    configLoaded: boolean
+}
+
+let defaultContext: ConfigContextInterface = {
+    config: undefined,
+    setConfig: () => {},
+    configLoaded: false
+}
+
+export const ConfigContext = createContext<ConfigContextInterface>(defaultContext);
+
+export const ConfigContextProvider: React.FunctionComponent = ({children}) => {
+    const [config, setConfig] = useState<Config | undefined>();
+    const save = useRef(false);
+
+    useEffect(() => {
+        getConfig().then(setConfig);
+    }, [])
+
+    // Tell the backend to store a new config when it changes
+    // Save ref is used to not post the first request
+    useEffect(() => {
+        if (config) {
+            if (save.current) {
+                postConfig(config);
```

Comment:
If postConfig fails for some reason it would be great to have an error and roll back to the last config.

It's a bit ugly but The current code base uses alertUser component maybe good enough for now?

Reply:
I changed the code, 
Now, on `set`, it gets the new value from the server, updates the config with that value, pushes the new config, and if it fails restores it to the latest gotten from the server, or the latest known if that request failed